modules/cloudsql-postgres/main.tf

```diff
@@ -0,0 +1,122 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+    }
+  }
+}
+
+# Labels
+
+locals {
+  default_labels = {
+    cloudsql = var.name
+  }
+
+  squad_label = var.squad != "" ? {
+    squad = var.squad
+    team  = var.squad
+  } : {}
+
+  merged_labels = merge(local.default_labels, local.squad_label)
+}
+
+# Primary Cloud SQL instance
+
+resource "google_sql_database_instance" "this" {
+  name    = var.name
+  project = var.project
+  region  = var.region
+
+  database_version    = var.database_version
+  deletion_protection = var.deletion_protection
+
+  settings {
+    tier                  = var.tier
+    availability_type     = var.enable_high_availability ? "REGIONAL" : "ZONAL"
+    disk_type             = "PD_SSD"
+    disk_size             = var.storage_gb
+    disk_autoresize       = true
+    disk_autoresize_limit = var.disk_autoresize_limit
+
+    ip_configuration {
+      ipv4_enabled    = false
+      private_network = var.network
+    }
+
+    backup_configuration {
+      enabled                        = var.backup_enabled
+      start_time                     = var.backup_start_time
+      point_in_time_recovery_enabled = var.enable_point_in_time_recovery
```

Comment:
documentation here suggests `point_in_time_recovery_enabled` should be set to `true` for postgres

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#availability_type-1

Reply:
can we do something smarter where if running with master/replicas, we check that these properties are set to true so?